src/app/components/LinkButton/LinkButton.tsx

```diff
@@ -1,22 +1,28 @@
+import classNames from 'classnames';
 import Link, { LinkProps } from 'next/link';
 
 interface LinkButtonProps extends LinkProps {
   children: React.ReactNode;
   variant: 'primary' | 'secondary';
+  className?: string;
   disabled?: boolean;
 }
 export function LinkButton({
   children,
   variant,
   href,
+  className = '',
   disabled = false,
   ...props
 }: LinkButtonProps) {
   if (variant === 'secondary') {
     return (
       <Link
         href={href}
-        className="usa-button hover:bg-red items-center justify-start gap-2.5 rounded border-2 border-[#224a58] bg-white px-5 py-3 hover:border-2"
+        className={classNames(
+          'usa-button hover:bg-red items-center gap-2.5 rounded border-2 border-[#224a58] bg-white px-5 py-3 hover:border-2',
```

Comment:
We should refactor these class name strings into data objects. Then we can just set the classes with that. 

So we can have a disabled button return,

then a another one, and set the classes based on if the variant is secondary

Reply:
Can you give me an example of what you envision the "data object" to look like? Just want to make sure I get what you mean.

Reply:
```
import classNames from 'classnames';
import Link, { LinkProps } from 'next/link';

const BUTTON_STYLES = {
  primary: {
    button: 'usa-button bg-[#224a58] hover:bg-green',
    text: 'text-white',
  },
  secondary: {
    button:
      'usa-button hover:bg-red items-center gap-2.5 rounded border-2 border-[#224a58] bg-white px-5 py-3 hover:border-2',
    text: 'text-[#224a58]',
  },
} as const;

interface LinkButtonProps extends LinkProps {
  children: React.ReactNode;
  variant: 'primary' | 'secondary';
  className?: string;
  disabled?: boolean;
}
interface LinkButtonProps extends LinkProps {
  children: React.ReactNode;
  variant: 'primary' | 'secondary';
  className?: string;
  disabled?: boolean;
  'aria-label'?: string;
}
export function LinkButton({
  children,
  variant,
  href,
  className = '',
  disabled = false,
  'aria-label': ariaLabel,
  ...props
}: LinkButtonProps) {
  if (disabled) {
    return (
      <div
        className="usa-button usa-button--disabled self-end justify-self-start bg-[#224a58]"
        aria-disabled="true"
        role="link"
        aria-label={ariaLabel}
      >
        <span className="text-base font-bold text-black">{children}</span>
      </div>
    );
  }

  return (
    <Link
      href={href}
      className={classNames(BUTTON_STYLES[variant].button, className)}
      aria-label={ariaLabel}
      {...props}
    >
      <span
        className={classNames(
          'text-center text-base font-bold',
          BUTTON_STYLES[variant].text,
        )}
      >
        {children}
      </span>
    </Link>
  );
}
```

Reply:
I like it. Thanks!